diff -ru telephony-kmods-a/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/wctc4xxp/base.c telephony-kmods-b/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/wctc4xxp/base.c
--- telephony-kmods-a/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/wctc4xxp/base.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/wctc4xxp/base.c	2026-06-17 00:00:00.000000000 +0000
@@ -3652,9 +3652,9 @@
 }
 
 static void
-wctc4xxp_watchdog(TIMER_DATA_TYPE timer)
+wctc4xxp_watchdog(struct timer_list *timer)
 {
-	struct wcdte *wc = from_timer(wc, timer, watchdog);
+	struct wcdte *wc = container_of(timer, struct wcdte, watchdog);
 	struct tcb *cmd, *temp;
 	LIST_HEAD(cmds_to_retry);
 	const int MAX_RETRIES = 5;
diff -ru telephony-kmods-a/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/wcte12xp/base.c telephony-kmods-b/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/wcte12xp/base.c
--- telephony-kmods-a/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/wcte12xp/base.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/wcte12xp/base.c	2026-06-17 00:00:00.000000000 +0000
@@ -2777,10 +2777,10 @@
 	return;
 }
 
-static void te12xp_timer(TIMER_DATA_TYPE timer)
+static void te12xp_timer(struct timer_list *timer)
 {
 	unsigned long flags;
-	struct t1 *wc = from_timer(wc, timer, timer);
+	struct t1 *wc = container_of(timer, struct t1, timer);
 
 	if (unlikely(!test_bit(INITIALIZED, &wc->bit_flags)))
 		return;
diff -ru telephony-kmods-a/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/xpp/xbus-sysfs.c telephony-kmods-b/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/xpp/xbus-sysfs.c
--- telephony-kmods-a/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/xpp/xbus-sysfs.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/xpp/xbus-sysfs.c	2026-06-17 00:00:00.000000000 +0000
@@ -397,7 +397,7 @@
 ATTRIBUTE_GROUPS(xbus_dev);
 #endif
 
-static int astribank_match(struct device *dev, struct device_driver *driver)
+static int astribank_match(struct device *dev, const struct device_driver *driver)
 {
 	DBG(DEVICES, "SYSFS MATCH: dev->bus_id = %s, driver->name = %s\n",
 	    dev_name(dev), driver->name);
@@ -771,7 +771,7 @@
 	return len;
 }
 
-static int xpd_match(struct device *dev, struct device_driver *driver)
+static int xpd_match(struct device *dev, const struct device_driver *driver)
 {
 	struct xpd_driver *xpd_driver;
 	xpd_t *xpd;
diff -ru telephony-kmods-a/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/xpp/xbus-pcm.c telephony-kmods-b/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/xpp/xbus-pcm.c
--- telephony-kmods-a/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/xpp/xbus-pcm.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/xpp/xbus-pcm.c	2026-06-17 00:00:00.000000000 +0000
@@ -350,9 +350,9 @@
 			 (syncer) ? syncer->busname : "NO-SYNC");
 }
 
-static void xbus_command_timer(TIMER_DATA_TYPE timer)
+static void xbus_command_timer(struct timer_list *timer)
 {
-	xbus_t *xbus = from_timer(xbus, timer, command_timer);
+	xbus_t *xbus = container_of(timer, xbus_t, command_timer);
 
 	BUG_ON(!xbus);
 	xbus_command_queue_tick(xbus);
