diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/dahdi-base.c dahdi-linux-complete-3.3.0+3.3.0-b/linux/drivers/dahdi/dahdi-base.c
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/dahdi-base.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/dahdi-base.c	2023-01-17 12:04:18.322695859 +0000
@@ -82,8 +82,8 @@
 
 #include "hpec/hpec_user.h"
 
-/* Linux kernel 5.16 and greater has removed user-space headers from the kernel include path */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 16, 0)
+/* Linux kernel 5.14 and greater has removed user-space headers from the kernel include path */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
 #include <asm/types.h>
 #else
 #include <stdbool.h>
diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/voicebus/vpmoct.h dahdi-linux-complete-3.3.0+3.3.0-b/linux/drivers/dahdi/voicebus/vpmoct.h
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/voicebus/vpmoct.h	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/voicebus/vpmoct.h	2023-01-17 12:05:24.898090118 +0000
@@ -30,8 +30,8 @@
 #include <linux/timer.h>
 #include "dahdi/kernel.h"
 
-/* Linux kernel 5.16 and greater has removed user-space headers from the kernel include path */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 16, 0)
+/* Linux kernel 5.14 and greater has removed user-space headers from the kernel include path */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
 #include <asm/types.h>
 #else
 #include <stdbool.h>
diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/wcaxx-base.c dahdi-linux-complete-3.3.0+3.3.0-b/linux/drivers/dahdi/wcaxx-base.c
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wcaxx-base.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wcaxx-base.c	2023-01-17 12:06:40.423537377 +0000
@@ -34,8 +34,8 @@
 #include <linux/firmware.h>
 #include <linux/crc32.h>
 
-/* Linux kernel 5.16 and greater has removed user-space headers from the kernel include path */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 16, 0)
+/* Linux kernel 5.14 and greater has removed user-space headers from the kernel include path */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
 #include <asm/types.h>
 #else
 #include <stdbool.h>
diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/wct4xxp/base.c dahdi-linux-complete-3.3.0+3.3.0-b/linux/drivers/dahdi/wct4xxp/base.c
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wct4xxp/base.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wct4xxp/base.c	2023-01-17 12:07:07.844699764 +0000
@@ -42,8 +42,8 @@
 #include <linux/crc32.h>
 #include <linux/slab.h>
 
-/* Linux kernel 5.16 and greater has removed user-space headers from the kernel include path */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 16, 0)
+/* Linux kernel 5.14 and greater has removed user-space headers from the kernel include path */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
 #include <asm/types.h>
 #else
 #include <stdbool.h>
diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/wct4xxp/vpm450m.c dahdi-linux-complete-3.3.0+3.3.0-b/linux/drivers/dahdi/wct4xxp/vpm450m.c
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wct4xxp/vpm450m.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wct4xxp/vpm450m.c	2023-01-17 12:07:42.161902989 +0000
@@ -29,8 +29,8 @@
 
 #include <dahdi/kernel.h>
 
-/* Linux kernel 5.16 and greater has removed user-space headers from the kernel include path */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 16, 0)
+/* Linux kernel 5.14 and greater has removed user-space headers from the kernel include path */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
 #include <asm/types.h>
 #else
 #include <stdbool.h>
diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/wctc4xxp/base.c dahdi-linux-complete-3.3.0+3.3.0-b/linux/drivers/dahdi/wctc4xxp/base.c
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wctc4xxp/base.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wctc4xxp/base.c	2023-01-17 12:08:58.352354186 +0000
@@ -40,8 +40,8 @@
 #include <linux/timer.h>
 #include <dahdi/kernel.h>
 
-/* Linux kernel 5.16 and greater has removed user-space headers from the kernel include path */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 16, 0)
+/* Linux kernel 5.14 and greater has removed user-space headers from the kernel include path */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
 #include <asm/types.h>
 #else
 #include <stdbool.h>
diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/wctdm24xxp/base.c dahdi-linux-complete-3.3.0+3.3.0-b/linux/drivers/dahdi/wctdm24xxp/base.c
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wctdm24xxp/base.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wctdm24xxp/base.c	2023-01-17 12:09:23.762504664 +0000
@@ -54,8 +54,8 @@
 #include <linux/crc32.h>
 #include <linux/slab.h>
 
-/* Linux kernel 5.16 and greater has removed user-space headers from the kernel include path */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 16, 0)
+/* Linux kernel 5.14 and greater has removed user-space headers from the kernel include path */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
 #include <asm/types.h>
 #else
 #include <stdbool.h>
diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/wcte13xp-base.c dahdi-linux-complete-3.3.0+3.3.0-b/linux/drivers/dahdi/wcte13xp-base.c
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wcte13xp-base.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wcte13xp-base.c	2023-01-17 12:09:50.183661130 +0000
@@ -36,8 +36,8 @@
 #include <linux/crc32.h>
 #include <dahdi/kernel.h>
 
-/* Linux kernel 5.16 and greater has removed user-space headers from the kernel include path */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 16, 0)
+/* Linux kernel 5.14 and greater has removed user-space headers from the kernel include path */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
 #include <asm/types.h>
 #else
 #include <stdbool.h>
diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/wcte43x-base.c dahdi-linux-complete-3.3.0+3.3.0-b/linux/drivers/dahdi/wcte43x-base.c
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wcte43x-base.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wcte43x-base.c	2023-01-17 12:10:10.378780724 +0000
@@ -43,8 +43,8 @@
 #include <linux/firmware.h>
 #include <oct612x.h>
 
-/* Linux kernel 5.16 and greater has removed user-space headers from the kernel include path */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 16, 0)
+/* Linux kernel 5.14 and greater has removed user-space headers from the kernel include path */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
 #include <asm/types.h>
 #else
 #include <stdbool.h>
diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/wcxb.c dahdi-linux-complete-3.3.0+3.3.0-b/linux/drivers/dahdi/wcxb.c
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wcxb.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wcxb.c	2023-01-17 12:10:26.880878451 +0000
@@ -36,8 +36,8 @@
 
 #include <dahdi/kernel.h>
 
-/* Linux kernel 5.16 and greater has removed user-space headers from the kernel include path */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 16, 0)
+/* Linux kernel 5.14 and greater has removed user-space headers from the kernel include path */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
 #include <asm/types.h>
 #else
 #include <stdbool.h>
diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/wcxb_spi.h dahdi-linux-complete-3.3.0+3.3.0-b/linux/drivers/dahdi/wcxb_spi.h
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wcxb_spi.h	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/wcxb_spi.h	2023-01-17 12:10:41.896967376 +0000
@@ -26,8 +26,8 @@
 #include <linux/spi/spi.h>
 #include <linux/version.h>
 
-/* Linux kernel 5.16 and greater has removed user-space headers from the kernel include path */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 16, 0)
+/* Linux kernel 5.14 and greater has removed user-space headers from the kernel include path */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
 #include <asm/types.h>
 #else
 #include <stdbool.h>

diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/xpp/xbus-core.c dahdi-linux-complete-3.3.0+3.3.0-b/linux/drivers/dahdi/xpp/xbus-core.c
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/xpp/xbus-core.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/xpp/xbus-core.c	2023-01-19 15:04:55.465060629 +0000
@@ -1839,7 +1839,7 @@
 
 static int xbus_read_proc_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, xbus_proc_show, PDE_DATA(inode));
+	return single_open(file, xbus_proc_show, pde_data(inode));
 }
 
 #ifdef DAHDI_HAVE_PROC_OPS
@@ -1946,7 +1946,7 @@
 
 static int proc_xbus_command_open(struct inode *inode, struct file *file)
 {
-	file->private_data = PDE_DATA(inode);
+	file->private_data = pde_data(inode);
 	return 0;
 }
 
@@ -1989,7 +1989,7 @@
 
 static int xpp_proc_read_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, xpp_proc_read_show, PDE_DATA(inode));
+	return single_open(file, xpp_proc_read_show, pde_data(inode));
 }
 
 #ifdef DAHDI_HAVE_PROC_OPS

diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/include/dahdi/kernel.h dahdi-linux-complete-3.3.0+3.3.0-b/linux/include/dahdi/kernel.h
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/include/dahdi/kernel.h	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/include/dahdi/kernel.h	2023-01-25 16:39:54.152534164 +0000
@@ -1488,7 +1488,7 @@
 #endif /* 4.15.0 */
 #endif /* 5.6 */
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 17, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
 #ifdef CONFIG_PROC_FS
 #define PDE_DATA(i)	pde_data(i)
 #endif

diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/xpp/xbus-sysfs.c dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/xpp/xbus-sysfs.c
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/xpp/xbus-sysfs.c	2024-05-13 13:09:03.436514200 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/xpp/xbus-sysfs.c	2024-05-13 13:35:05.463048037 +0100
024-05-14 10:57:37.780153149 +0100
@@ -420,7 +420,7 @@
 
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(6, 3, 0)
-static int astribank_uevent(struct device *dev, struct kobj_uevent_env *kenv)
+static int astribank_uevent(const struct device *dev, struct kobj_uevent_env *kenv)
 #else
 static int astribank_uevent(const struct device *dev, struct kobj_uevent_env *kenv)
 #endif

diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/dahdi-sysfs.c dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/dahdi-sysfs.c
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/dahdi-sysfs.c	2024-05-13 13:09:03.436514200 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/dahdi-sysfs.c	2024-05-13 13:35:05.463048037 +0100
@@ -69,7 +69,7 @@
 	} while (0)
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(6, 3, 0)
-static int span_uevent(struct device *dev, struct kobj_uevent_env *kenv)
+static int span_uevent(const struct device *dev, struct kobj_uevent_env *kenv)
 #else
 static int span_uevent(const struct device *dev, struct kobj_uevent_env *kenv)
 #endif
@@ -444,7 +444,7 @@
 
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(6, 3, 0)
-static int device_uevent(struct device *dev, struct kobj_uevent_env *kenv)
+static int device_uevent(const struct device *dev, struct kobj_uevent_env *kenv)
 #else
 static int device_uevent(const struct device *dev, struct kobj_uevent_env *kenv)
 #endif

diff -ru dahdi-linux-complete-3.3.0+3.3.0-a/linux/drivers/dahdi/dahdi-sysfs-chan.c dahdi-linux-complete-3.3.0+3.3.0-/linux/drivers/dahdi/dahdi-sysfs-chan.c
--- telephony-kmods-a/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/dahdi-sysfs-chan.c	2024-05-14 08:49:25.122784694 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.3.0+3.3.0/linux/drivers/dahdi/dahdi-sysfs-chan.c	2024-05-14 11:23:04.085816566 +0100
@@ -484,7 +484,7 @@
 	should_cleanup.channel_driver = 1;
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0)
-	dahdi_class = class_create(THIS_MODULE, "dahdi");
+	dahdi_class = class_create("dahdi");
 #else
 	dahdi_class = class_create("dahdi");
 #endif
