diff -ru dahdi-linux-complete-3.4.0+3.4.0-a/linux/drivers/dahdi/wctc4xxp/base.c dahdi-linux-complete-3.4.0+3.4.0-b/linux/drivers/dahdi/wctc4xxp/base.c
--- telephony-kmods-a/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/wctc4xxp/base.c	2022-09-22 13:23:24.000000000 +0100
+++ telephony-kmods-b/dahdi-linux-complete-3.4.0+3.4.0/linux/drivers/dahdi/wctc4xxp/base.c	2023-05-18 14:37:24.918334554 +0100
@@ -658,7 +658,7 @@
 	netdev->promiscuity = 0;
 	netdev->flags |= IFF_NOARP;

-	netif_napi_add(netdev, &wc->napi, &wctc4xxp_poll, 64);
+	netif_napi_add_weight(netdev, &wc->napi, &wctc4xxp_poll, 64);

 	res = register_netdev(netdev);
 	if (res) {
