diff --no-dereference -ruw wanpipe-7.0.38-a/patches/kdrivers/src/net/wanpipe_cdev_linux.c wanpipe-7.0.38/patches/kdrivers/src/net/wanpipe_cdev_linux.c
--- telephony-kmods-a/wanpipe-7.0.38/patches/kdrivers/src/net/wanpipe_cdev_linux.c	2025-05-26 09:30:15.507742440 +0100
+++ telephony-kmods-b/wanpipe-7.0.38/patches/kdrivers/src/net/wanpipe_cdev_linux.c	2025-05-26 12:27:13.158158961 +0100
@@ -211,7 +211,7 @@
 		return -EINVAL;
 	}

-	wp_cdev_class = class_create(THIS_MODULE, "wanpipe");
+	wp_cdev_class = class_create("wanpipe");
 	if (IS_ERR(wp_cdev_class)) {
 		DEBUG_ERROR("%s(): Error creating class!\n",__FUNCTION__);
 		cdev_del(&wp_cdev_dev);
diff --no-dereference -ruw wanpipe-7.0.38-a/patches/kdrivers/src/wanrouter/af_wanpipe.c wanpipe-7.0.38/patches/kdrivers/src/wanrouter/af_wanpipe.c
--- telephony-kmods-a/wanpipe-7.0.38/patches/kdrivers/src/wanrouter/af_wanpipe.c	2025-05-26 09:30:15.513002788 +0100
+++ telephony-kmods-b/wanpipe-7.0.38/patches/kdrivers/src/wanrouter/af_wanpipe.c	2025-05-26 12:13:58.936387265 +0100
@@ -572,11 +572,7 @@
  *	For each incoming call, create a new socket and
  *      return it to the user.
  *=====================================================================*/
-#if defined(KERN_TIMER_SETUP) && KERN_TIMER_SETUP > 0
-static int *wanpipe_accept(struct socket *sock, struct socket *newsock, int flags, bool kern)
-#else
-static int *wanpipe_accept(struct socket *sock, struct socket *newsock, int flags)
-#endif
+static int wanpipe_accept(struct socket *sock, struct socket *newsock,struct proto_accept_arg *arg)
 {
 	struct sock *sk;
 	struct sock *newsk=NULL, *newsk_frmskb=NULL;

diff -ruw wanpipe-7.0.38-a/patches/kdrivers/src/wanrouter/af_wanpipe_src.c wanpipe-7.0.38/patches/kdrivers/src/wanrouter/af_wanpipe_src.c
--- telephony-kmods-a/wanpipe-7.0.38/patches/kdrivers/src/wanrouter/af_wanpipe_src.c	2025-05-26 09:30:15.513002788 +0100
+++ telephony-kmods-b/wanpipe-7.0.38/patches/kdrivers/src/wanrouter/af_wanpipe_src.c	2025-05-26 12:13:58.936387265 +0100
@@ -572,11 +572,7 @@
  *	For each incoming call, create a new socket and
  *      return it to the user.
  *=====================================================================*/
-#if defined(KERN_TIMER_SETUP) && KERN_TIMER_SETUP > 0
-static int *wanpipe_accept(struct socket *sock, struct socket *newsock, int flags, bool kern)
-#else
-static int *wanpipe_accept(struct socket *sock, struct socket *newsock, int flags)
-#endif
+static int wanpipe_accept(struct socket *sock, struct socket *newsock,struct proto_accept_arg *arg)
 {
 	struct sock *sk;
 	struct sock *newsk=NULL, *newsk_frmskb=NULL;
diff --no-dereference -ruw wanpipe-7.0.38-a/patches/kdrivers/wanec/wanec_dev.c wanpipe-7.0.38/patches/kdrivers/wanec/wanec_dev.c
--- telephony-kmods-a/wanpipe-7.0.38/patches/kdrivers/wanec/wanec_dev.c	2025-05-26 09:30:15.538032722 +0100
+++ telephony-kmods-b/wanpipe-7.0.38/patches/kdrivers/wanec/wanec_dev.c	2025-05-26 12:27:44.844309215 +0100
@@ -139,7 +139,7 @@

 	DEBUG_EVENT("%s: Registering Wanpipe ECDEV Device!\n",__FUNCTION__);
 #ifdef WP_ECDEV_UDEV
-	wanec_dev_class = class_create(THIS_MODULE, "wp_ec");
+	wanec_dev_class = class_create("wp_ec");
 #endif

 #ifdef WP_CONFIG_DEVFS_FS
