diff -ru wanpipe-7.0.36-a/patches/kdrivers/include/wanpipe_common.h wanpipe-7.0.36-b/patches/kdrivers/include/wanpipe_common.h
--- telephony-kmods-a/wanpipe-7.0.36/patches/kdrivers/include/wanpipe_common.h	2023-11-22 15:15:15.562673553 +0000
+++ telephony-kmods-b/wanpipe-7.0.36/patches/kdrivers/include/wanpipe_common.h	2023-11-22 15:17:27.216730640 +0000
@@ -798,7 +798,7 @@
 {
 #if defined(__LINUX__)
 # if  (LINUX_VERSION_CODE < KERNEL_VERSION(6,0,0))
-	return virt_to_bus(ptr);
+	return virt_to_phys(ptr);
 #else
 	return isa_virt_to_bus(ptr);
 #endif
@@ -821,7 +821,7 @@
 {
 #if defined(__LINUX__)
 # if  (LINUX_VERSION_CODE < KERNEL_VERSION(6,0,0))
-	return bus_to_virt(virt_addr);
+	return phys_to_virt(virt_addr);
 #else
 	return isa_bus_to_virt(virt_addr);
 #endif

diff -ru wanpipe-7.0.36-a/patches/kdrivers/include/wanpipe_kernel.h wanpipe-7.0.36-b/patches/kdrivers/include/wanpipe_kernel.h 
--- telephony-kmods-a/wanpipe-7.0.36/patches/kdrivers/include/wanpipe_kernel.h	2023-11-22 15:15:15.561673553 +0000
+++ telephony-kmods-bwanpipe-7.0.36/patches/kdrivers/include/wanpipe_kernel.h	2023-11-22 15:52:06.239512467 +0000
@@ -287,7 +287,7 @@
  #define dev_init_buffers(a)
 
  #if defined(KERN_PROC_PDE_FEATURE) && KERN_PROC_PDE_FEATURE > 0
- #if LINUX_VERSION_CODE >= KERNEL_VERSION(5,17,0)
+ #if LINUX_VERSION_CODE >= KERNEL_VERSION(5,14,0)
  # define WP_PDE_DATA pde_data
  #else
  # define WP_PDE_DATA PDE_DATA
