diff -ru wanpipe-7.0.34-a/patches/kdrivers/include/wanpipe_kernel.h wanpipe-7.0.34-b/patches/kdrivers/include/wanpipe_kernel.h
--- telephony-kmods-a/wanpipe-7.0.34/patches/kdrivers/include/wanpipe_kernel.h	2008-08-01 14:00:00.000000000 +0100
+++ telephony-kmods-b/wanpipe-7.0.34/patches/kdrivers/include/wanpipe_kernel.h	2023-01-25 16:56:03.517431918 +0000
@@ -286,15 +286,20 @@
 
  #define dev_init_buffers(a)
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
+#define PDE_DATA(i)	pde_data(i)
+#endif
+
+
  #if defined(KERN_PROC_PDE_FEATURE) && KERN_PROC_PDE_FEATURE > 0
- # define WP_PDE_DATA PDE_DATA
+ # define WP_PDE_DATA pde_data
  #else 
  #include <linux/proc_fs.h>
  static inline void*WP_PDE_DATA(const struct inode *inode)
  {	
-	struct proc_dir_entry* pde=PDE(inode);
+	struct proc_dir_entry* pde=PDE_DATA(inode);
 	if (pde) {
-		return pde->data;
+		return PDE_DATA(inode);
 	}	
     
 	return NULL;

--- telephony-kmods-a/wanpipe-7.0.34/patches/kdrivers/include/wanpipe_wanrouter.h	2008-08-01 14:00:00.000000000 +0100
+++ telephony-kmods-b/wanpipe-7.0.34/patches/kdrivers/include/wanpipe_wanrouter.h	2023-01-26 15:29:07.093000835 +0000
@@ -336,8 +336,6 @@
 	
 	wan_get_info_t*	get_dev_config_info;
 	wan_get_info_t*	get_if_info;
-	write_proc_t*	set_dev_config;
-	write_proc_t*	set_if_info;
 #endif
 
 	int 	(*get_info)(void*, struct seq_file* m, int *);
