diff -ru wanpipe-7.0.38-a/patches/kdrivers/src/net/wanpipe_cdev_linux.c wanpipe-7.0.38/patches/kdrivers/src/net/wanpipe_cdev_linux.c
--- telephony-kmods-a/wanpipe-7.0.38/patches/kdrivers/src/net/wanpipe_cdev_linux.c	2024-05-14 11:36:31.284390203 +0100
+++ telephony-kmods-b/wanpipe-7.0.38/patches/kdrivers/src/net/wanpipe_cdev_linux.c	2024-05-14 11:33:16.444527674 +0100
@@ -211,14 +211,14 @@
 		return -EINVAL;
 	}

-	wp_cdev_class = class_create(THIS_MODULE, "wanpipe");
+	wp_cdev_class = class_create("wanpipe");
 	if (IS_ERR(wp_cdev_class)) {
 		DEBUG_ERROR("%s(): Error creating class!\n",__FUNCTION__);
 		cdev_del(&wp_cdev_dev);
 		unregister_chrdev_region(wanpipe_dev, WP_CDEV_MAX_MINORS);
 		return -EINVAL;
 	}
-
+
 	err = bus_register(&wanpipe_device_bus);
     if (err) {
 		DEBUG_ERROR("%s(): Error registering bus!\n",__FUNCTION__);
